src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialProvider.java

```diff
@@ -146,6 +147,15 @@ public void stopWatchingForSecrets() {
         return Collections.emptyList();
     }
 
+    /**
+     * Gets the singleton instance.
+     *
+     * @return the singleton instance.
+     */
+    public static KubernetesCredentialProvider getInstance() {
+        return ExtensionList.lookup(KubernetesCredentialProvider.class).get(KubernetesCredentialProvider.class);
```

Comment:
see `ExtensionList.lookupSingleton()`
```suggestion
        return ExtensionList.lookupSingleton(KubernetesCredentialProvider.class);
```